src/faq.md

```diff
@@ -79,7 +79,62 @@ outputs:
       outputEval: ${self[0].basename=inputs.otu_table_name; return self;}
 ```
 
-## Setting `self`-based input bindings for optional inputs
+## Referencing a Local Script
+
+There are two ways to reference a local script:
+
+The first method involves adding the folder containing your scripts to the `PATH` environment variable. This allows you to run the shell script without using `sh` or `bash` commands.
+
+Start with adding a _shebang_ at the top of your file:
+
+```{code-block}
+#!/bin/bash
+```
+
+After that, make the script executable with the command `chmod +x scriptname.sh`
+
+Finally, modify your `PATH` to add the directory where your script is located.
+(It is good practice to use `$HOME/bin` for storing your own scripts).
+
+```bash
+export PATH=$PATH:/appropriate/directory
```

Comment:
> > Also, the commit messages are very uninformative. Either PRs should be squashed and the commit message updated, or contributors should be asked to clean up their commits.
> 
> Did we give examples in the contributions guide? If not - perhaps we should since I didn't know we had strict rules about commit messages. Perhaps you should give an examples in an attempt to help people who are new to github or writing commit messages. I would vote for squashing because I believe most of the contributors did this during the contributions period and it might be hard to circle back with them in a reasonable time frame.

https://github.com/common-workflow-language/user_guide/blob/main/CONTRIBUTING.md#using-github links to https://docs.github.com/en/get-started/quickstart/github-flow which contains

> Give each commit a descriptive message to help you and future contributors understand what changes the commit contains. For example, fix typo or increase rate limit.
>
> Ideally, each commit contains an isolated, complete change. This makes it easy to revert your changes if you decide to take a different approach. For example, if you want to rename a variable and add some tests, put the variable rename in one commit and the tests in another commit. Later, if you want to keep the tests but revert the variable rename, you can revert the specific commit that contained the variable rename. If you put the variable rename and tests in the same commit or spread the variable rename across multiple commits, you would spend more effort reverting your changes.

Note that this norm is hardly unique to the CWL User Guide (including [the Arvados project](https://dev.arvados.org/projects/arvados/wiki/Coding_Standards#Git-commits)), so I would expect all mergers of PRs to either squash the PR down to one commit with a useful message (for smaller changes) or work with the contributor to reformat the commits and merge/rebase them.

[I don't know why your comment was left as a reply to an unrelated change; it makes the conversation hard to follow]

Reply:
I am not saying it is not a "norm" for development projects, I was just wanted to make sure that this was mentioned somewhere in the contributing guide.  If we think it is a big issue, it might be good to put it directly in the guide instead of hoping they will read the link and find it in the link.

Reply:
I'll create an issue to add the contributing guide with instructions about squashing commits. Marking this thread as resolved.